core/src/main/scala/chisel3/Data.scala

```diff
@@ -1089,7 +1056,7 @@ object WireDefault {
   ): T = {
     val x = Wire(t)
     requireIsHardware(init, "wire initializer")
-    x := init
+    x.asInstanceOf[Data] :#= init
```

Comment:
Hm, why doesn't this take T as init?

Reply:
Maybe to initialize it to don't care? I'm not sure..